### ISSUE TITLE
Convert patients/care-goals/AddCareGoalModal.test.tsx to RTL

### ISSUE THREAD

**Issue by Proful** [kcdraidgroup/hospitalrun-frontend#25]
`expect(screen.getByRole('form')).toBeInTheDocument()`

`TestingLibraryElementError: Unable to find an accessible element with the role "form"`

https://github.com/testing-library/dom-testing-library/issues/474

> This is actually intended by spec. A <form> element needs an accessible name to have the form role. So you need <form aria-label="form" /> or <form aria-labelledby="form-label-element-id" />.

```
<div
  class="modal-body"
>
  <form>
    <div
      class="row"
    >  
```
how to test form is present or not?

`expect(container.querySelector('form')).toBeInTheDocument()`  getting null

Need help

**Comment by balavishnuvj** [comment]
does `expect(screen..querySelector('form')).toBeInTheDocument()` does anything different?

**Comment by Proful** [comment]
screen don't have querySelector 😊

**Comment by Proful** [comment]
Q is it possible to add some aria-label to form elements

> feel free to update that code to add an aria-label!
-[Jack](https://discord.com/channels/715220730605731931/785649901782433852/789319835775402004)

**Comment by Proful** [comment]
[playground](https://testing-playground.com/#markup=DwEwlgbgBAhgTmGBaAFmEICmA7AvAIgBc4BXTfAPmAHpwIq6oBjAGxgGd2CAzGLKALYB7EDBZIARjCYBrEHCEAHKOxRCA7pRp0GkKApaYC4MUIDm+WAmTDRLAsTKXWHLvl79bYlWs1RCMBJg2FgAHgRIAIyW7IQAnobGYOyKbHEAXFASLEKyANxajAZG+CC5JAI4hM5snARe4iY5FrrQLnX4DUhMQtiEVYV67W5dKJh8mHCDbbUjImJIhGCEhlAoACyUijBLVQB0TPCYAOJCYnvYmOrakFQSJISEvf5xiiX3j701rgSsQuzkKgpGDYKyIVDoLB4IikQEAdZowOwQO2oOGBHYcCQvRYcUoAGEcgDEaiqNQPk9kTd6KAhrN6vNxBIRHiqNwhHABGDkGwJJh7PhDnBMEgzGdxOzOdNmPT8ApNK0ZT9BUJxOwBFEAEzS9HuDkaswKEiKaZUXn8pUdSUa80sSyS4yYdhMBCKJa9AAqmFChAAYmB+SBLEsViUPWh2FBkrB9JgAI4kMDCkBR7CKB5bHYBvoHI6nc5YZ2u93YYBgFTxRIqnJwTLJgpAiBmbkQjA4Byw+3ldiBKu8FgAyyiAJIRTC7hgcLuDhDnbIMA9aEcfoIdgyb4ddhNpDBFjBEVIXhQXhIZeTZIyY-IdRRABsUEsxQIYAEFigoQELGwbhQj0U6Woah1GAvZ1AAZj2DkzGoTUAAZ4OoLc3wgAN1AAISEKdYKgbCAFZIk1KB8O1CsEhKMYwDMX9MgAdkUUI8igCBJiWQ5xDESjsEyJ5FEYgR4DMYIkEMbhCEyTV6IbYBtkIFBjzAFgBSYEg4GFPp8VVDkhwIABZdYaIADj2TVIigMCwPWPZYIATjAgAZcz701XDbxYSJ1k1YzIiQQyrNsphIkiPYaOs3CkFvPZ9JoqBIlwvZrNvbyTL2MCDPWKBrOMsDNSQcyrLAmjxEiTL1gM3KIJo9ZIiYcLgs1MrAr2W9cLApBkponLYsi6zrPK4zNRopAjNwzV7M1BKYoMiKaMiAyWAg-SwsiW9PLg9Z8XGzKaNvMCYqC2DEqgeqjNs6LsIGuLSpwlBcpO28asiBaXJwtrLPq3bGoOpKIJ69LNUs-D1jskz71m6bZqgGiaOM5qYtgzzb1gmiHqC3CDtq6ztraiL1h63K9kevrNQGqBhtGoqSoMsyKqqmrpvqvampatq9hcwjNQi57MtG3a8tggq7MibajuevKQupyybLAlHgtCqAcahqI4oS0zktS9YkG57Lqfywrisiqmxdp+mqca5rdvakyYsunqdeJ6KybG5azOc4yDM1cQFpo3CYpW4zYPWkzVoKoj4PisDrKO0arNvdLiKOsDOcI4iUFS+LEoCp773Ot6DNaz7Dujnqwv+1n3OBibsri+qPceoyaLAn20oJwKAtRg75fi4WOe6yOIMeu2BqG1nyf10q+sq6qTaiILzaO1mdux1ncN6rWcr5xOAC8tGoGSUDJJCyTAMlbTJHRaRmZVrVFI0TSofofSOGB9A0NxNktNxr8XYhVR3h+AmFDARUupggQA4iAQ8mBMAgCkLIHe59aC3GpGfJBiCaSMF1PKHUsoehqg1ARbBV99Q3yEMaU0wBbQfx4MQ20XY4AEDHGADkyw4gAGV+SYCYNUfwywqzhmjNGZ+woExJmgamdM1QKAyWzIQXMwp8wsD2Iw5h8Q9i2jLKRKsuCOR1mgVJJCLY0BtmhI4cgx5uy9hKP2QcUBhzIDHJgCcU5eDsFnCOBcvQCBnlXOuKh+AkI7mwHuS4SBDzPxPN4i8V4kA3mdo+VUJQXxvg-F+H8f4AJARAuBSCcBoJwQQkhSwKErgYSwjhIiBEKkkViGRAgFEqJiUhpJJiLEFwLA4mYLi-glB8QEkJESjSJIMS0HveSilfgqTUoQDSNZtL4D0r5K25lJa2QconEWrl3KrW8r5KWbdZZhQVtFLqKs2pBXVhlLK68IL80KlESmE9aa1Q6g1WezNLZK17kTIejsK6gymsFWa81IrezvKtAOG0erBR2ozL6R0DInRDudb2Btrq3XTg9LOL1S7vVhYlNqP1rJ-QBuXEGk1wZUyhjDH2kR4ZNSRjLNGt4MZYx7rjXq-dWp5XtqTEeYEKaoqNtVZ5DMzbvIXuzTm94ta8xuQLIWTlRYVT7hZPy0sZ4HM7lFT5KsjrnLSprK5OtbluUpjrSeTATaMznpbUyXV2WDwdny4Gzsq5uw9vjRavtwWBwIllaKaNYLh0jsTG5scqkJyTlU1Od1qqPUis9HObt877ULoSkuJKgbjXvG6mubk04NybpZQKcb273gipjJyONbacsdcPEa-Kx5lSFZauqrymYWwXq1HuuEV6GtGtc-Kt5t5kj3gfJsR8T6BH5Cg9BdJlRwAkNwmpVZSErH3JkbAvRMCMUUP8ZYTCunCjYEsZiBR-CBGCGECI0RFQrpKOAFIaRMjcEMAxY8b7MiREYvU6icNYIAFJd37pLLok9kAd1aGCBI2ADxchCAEKkTA-QCBCG4NwSwqRpCYDUCwLA9D8ChKgPiNQ-xMBQFCcGV4JR-6WHgOCGAcGeiIcMCh-AzJZJ0esEgb0qIsAgB4GIGx9HkAoA4Hu9MigCB7liMyUI8StEIeZHJjcbhF2EECRIpA-FgjmM5N0XoP8WD6CXZpjMTExBkAYVmfYQoTjiiUQgFRcQ9iVHABUKDaYHiGMhO2GETg-HqbMxptAfRHzjDKEEuIBAYiVneNIGQhpSEhAMzWbicAQTAimYxZkcB8OpZ0f4DL35tjZayJhJAqg+AaE3duxi2jaz6DMFIAAFLBAANDhTrHWcIpVwgASkYgMzIsFml7uCCuHjzE+jsFq5cYD7AD29EyIEdgqoHg7u6f+HCzT1DoFkl++CQHgyXpCN6G9RTLMlHgaghBd3btIIwbKLBwCcGqkqxqW8hCrTEKS2Q1oZpp3Gd1NfWhemMQBEICQdg7DDBcODLwsMEYoyRiEfGRMyZxEZikTZnMdmFF7FiDsaHGj72-E0o1+sWgDEidbFCDsAX2TKR7NkKxQmzF2NHOOScgnXG2LnDuRcXjYjnjXKpjE25dz7lCUeCJIufHRNifeBTiTXyWBSd+Op6TALAXUKBCCUEYLwVgohJsRTUKlIINhPClTiKxdqfgX9jS6LvuYnAVi7S9ydO4j0wQfTsDCUcYMySIydhyQnOMwUkyqgzK0rY3S+kjJLNVVLNZirNkeS8j5IyezGohUOSC45yt8VqzSpcgdfUTX3INo84V08xXdqCh1T57LvmDV+dm8lgK5peyWn7NakKtowoLqrBF4czpHRRVdWCN004JUxQm7Or1k14u+vFIlR1M0uv+RSyG0MOY0rpYjZGGqmUsuZWyvGnK2+8obQKq6LaRWmzed2iVR0pXl-Msa+V3clWywlmqjLPnlqorCcoXPqhrGvN-nrGai2lag3nqnVHajbCqv1E6g2tvi7NXO7J7CCjSv3hCkHP6qHEGrZCGtHAdHHJUqNFGinHPhnPGusImpvsmhqnCkXKFMvoDH8lge6vmvXI3KTMWq3I1Eyp3JWu-r3GZATLKmgfWqPA8nAW2hqjal2kvL2qvFcpXlvDvGOoiBOjQMfDQKfMghfIFkuvbqug8MEpgHNptnuotqBrGOBmeidkEGdlOFENKGTqUMkFhhkB+t6IxK+t6F+j+pgJRH+rSoBgtktkevyFmK4VQNBt5oxk8MxkhmxmhhhlAFhkwDhqqPhhEEgMRqRgCBRkgFRm8A4N6NwrTmkfBixshu8EIJxi2LxiCPxoJgOGYrTmJikEoMaNJskIQCpi-IpgIMpphOLnKKZikRpjpqCF-IZgYCZhpvMZdiwFZvgNIPkW6NAp5jBn0b5iYp2OYesV5iFhNuFtVlFjFpovFrIH9ilg1ulplqVlUDlhyPlq8UVu8UcH0F8aEJVmJmUOoHYfVhTnWM1jAG1t1vCb1o3INlAMNjtu+uNn0JMFNlULNlAFuvNrkSBoeithIGtlsf0IxDxCNrtvtigIdjEW4VeudoRtEBZlsddrOpyaYWgq9sqLgh9uFN9p-L9rfOQpQiDjQkDnQl4kwGgJgMxJUH0KwpDtDrDpwtwiGHwsjoIrGCIpjhsTjrsHjnmA5tIHKQqVUMqcTuwKTnFuTmlrGCAPok2D5sYgzmYkztDpYt0TYpzg4k4rzm4vOELjsfLheDMQElLiEmEtEpEmuIrneA+OMarskp+Jro7trpknrtkobvkiboUkxBbphFbuUvHHbg8XUhEQ0rRM0m7h7uxF7l0jxL0rkv0kHuJCHlQKMhHkpNHupBTnMgsknqZMsmqmnhsm5JniZNnoAXnpwUctbOnElBAZ-oOlXk2rXnTMoQ3mcnVC3lfhBPbPIWNBNGDN3sCotGCv7IHFCttB9KmqPoihPhdKijPuivPlEFirBMvriiPgSuvsSmXFms7GeTNJSvvrDLSgjAyqfujBWqytWhyjITfo7PfobDTHXm2tauKmzJIVzFlLIbcoLL-jmsqgAXshqsAUcjquASlAalARvDAYKhha2i8thRbE3lbPajWoeSTB3q6q7Hmp6t7N6teRtH6qNAGmHGQVHGGlQezInCPNGvQXGl+Swe9GwWmuvhmkBTwbmjgXXMFIIc3CWvsmIfBVWlIbWtyj8nyqajXkoS8ioS1PPC5N2pzH2mvNocOroWHuOmYJOsYUDlyU9guhYRWfgGujYRCYSY4cSc4YkZBheu4deiyd4Xab4U+jAAESEe+rlWEWsFWVEUdrEU4ceolVJPMbBukQhpkSUNkZhmwPkbhkUYRiUSRkIGRhUVUTRrUVxgxkxrVaxi0W0bTh0WdgJtOD0f1aJuJoMVJvgDJqMdMcmeTpMUIGMZgnMZcdpjALpssX0KsUFhsWydsbpmOOYMKJwIcakdxkYvTv5mYltRcVpqFtwoApFriPcT4bAolkaC8VCX8SVgCYQF8Xllib8cQP8WVnJiCdVuCXiXVswIDbkq1j1ujd1n1siaiaNuiUIBNlifKTiTFQ4XESSWSRtpSUoNSe+ntiAAdv+sdslUyZ4ayWAuyTFlyTydyfdnOpfB0C9mYbqPyeqIKbyT9vpn9nfBfIDnyMDrKKDlKeDv4gAoQAACI7CYAa39AAAKC4MgkwCOoYBA-CkYOpwiGOYiBp0itmJp5wRO7u2tmAtpDuvxVOjYzYxxbpj1XYzO3pU1vpAu-pPO04fOnOHiS4YZYufikZQS0uMZcuK4USJ4SuSZT4+ASS6uaZaShA-4OuWSBuuSRuBSZuhZJSxZ+A1uEa5ZPhTuNZrurSbEp4jZPuvEfurZAeqJQyUk3ZCkvZqkMeA58e8yieXk5FqyjkE5WyWeuy-kc5BeUUi5pype6UXljFaFm5T+zljee53FHKvF7ezqneoFQKveV5A+m00K95Mcj54+5SL50+s+d0C+TBS+OKecq+-5v0LB3BZKJ94F1KcM0FJ+ohcFXcF+iF+MhMNl6B5M1eD+LFT+7FLMuFPa0qBF0BxF6e5qKqKy6q89IBg0YBqsEB-a2s69Ta5qxsWFCBtqi5DqMDt+o0mB+lHqveolF9ElIcgawaslMc8lkaSldBsamci+99ucKaN9Cc2lXBpKlcglBlBaxlwhparMHcFlkhDDMhdaqFG5jl9U29rli8PanlWhG8Plo6fl+hAVhhU6ctIV86HQ8xJC-2QtsozjUtXOmAbwIQ1OqIfiHjt8iwfVHtrpD1pivtXprOPpHOQd3OziM4-O7iIZbEOAogWIYg3CuosdNhMu4SyAqTIQ8AzdhACZ7k0SkQoQadCSz4au742dWuudGSuu+uOSeSxupuyERZZS2E6wV0duXZYeYy-dUyseUww9OkvTscTA50nMiBPsV0izBkqcI0TAPa5S48yUCzZUpUAAasVJqAABKwQQADSaibwCDZTfnFT3TfkRT3iXRagxSdSagbCwStrZzPNESRRfMEQQDrDvPYT3NICXTPNPMEQ3QAtblL4EQguRRahagQBIAAsXPYQEQGQzO1QPPwuVKwsQtQv3PlK24-O4uaj-OAudzMqgt4uEQQvIvvNYsvSwuPM0tIsos6QHPv4Yt3NNTfMayktahvMfNEvJwku0tktQtAtNRwvpQ0vPOQsMuEvXM5QssvNsuwSovPPcuMuqtgv4vvNKtfPUvivkszOUsyt6visKvQtMsqs4tqv0ubw6ThoczauEu6s0tCuGvEuysmuSvmvGvgtHPbQ2vKsysIt-OOsCBov1SYvuv2uWteu8sxt8u-MSsUvAuBukuHMhuMthuquIuOt6TwTyzrCQsGT7O3PDxlT753h1vLSz6ag1RGRUzXP3i1sgW3j-MGSHPLSwSVuYsts+TGT1vMoNu5RNvVtMvtsjudvduHOlTiWsylvlIGRxQ+xot-OlQbClRsvas9q7lhTjzHsGQjo0B6GdMHykimGhUC3jBcJIDDjeP61YnqAZaKBvBTDi1qb3saZPuKAvtwAAD6QHzj38e1lwX7ZYlxLw1RRAfVfiB1RmSZ7N2xBEMEmoRucEN2NI3N3Nt7bg-8geokgg4gPsAgGmpkRH6oYgxmJAZ2E4lwQYXNvN379pApX2bHeoktopAOFCQOiHkpct0ppQZATtTtetsghtPCxt+AptKOMYFtoiKY1tuOsi+ODmIAYnmtLtimDp7twANOd1Jx7pkTLOfY7OQZXjAZodVnEdwuSd0d2Tkucd0ZsuyAcZl4KdiZKutTqZqSjTedWZrTuZHTBZxS6EFdVdZZBCEVddTSDd7ubSDZnErdLZgknd7ZR0nZ0kQzPZEyA9-ZsyEzo9yeeD45zkGe2yM5ueTe85heS9JeK5a9cqdyejiD9eL+u5zee9bex5PB-9F5oKy0Pqg+V9q+8KT599U+rbT9GKn5Yj357919X0X9G+pcv9IFAKYFe+gDUF9KIDZa5+S8re1+jDqF8D6FwUTy8BXXnkqDH+MqmDCqGyOD49+DdXPs1FxDiB6sZDhEFDsBLFt3HaiBHUyBVlB9TDTsOaCjbDeBHDhBXDUlpBEcfDlBEaNBQjELKlojr94jrBI+0jxcsjWa8j2BtcSjRaLcqj5l4DmjNa2jMDx59l48+j7aqhbl6hpjFe5jZ7u8Vjl7tjQV9jN7jjbggTpC0tBHz4O1njDiPjzHhn-jICcvQTtGoTXt4TZxnp5nbO01ST9i8TgZhvgunigoYgaTxTmTEZLnuTCdBTlvRTGTLApT3n5TJ4lT1TVYmd9TAXGZTT+d2Zhd7TJdXT5dPTUAfTVMAzuXskwzBXozQ9k1kzUf0zszy7WzUfVMSzKzTb6zvTZUWfJ7+z1kRzJzZzFzVz9yVb8b-L4rgrBLybRrYrzzprUrVLYrQb1rebCLBbkbKL0bWrcbvLHrLzSbnzPrab7fAbXfpLPfSrff9rhb7LnLrrI-2L9fibTfk-orvrbf-rmbc-Vr9LobS-9fK-GrQ-6LG-fLQbjfBrzfU-pLM-R-vr3fp-vfzLy-A-Grzrf0t4N1qPwTaesd+IrVNi-0P7Sss24rHNvdC-52sL+v-TVjfx5ab97++rYVim2NYH8M20A4-s8zgFn9v+SAslkWwBZAsy2yLCtstEHZ7Aa2s7MdkwMbbNt6B07SGIwN9jzs+2A7b8kOw7ZMC7wLAqdm2w4GdRQYXbUqAuwMhLsvu6UbCGu2XabsJWyzaPnu0xYHstm1A7QaVH54XtD4JIEEA435o-tpAf7TWgByk5Yg32MAD9obS46AIH2-7QDiBzA6GYIO9g6DjBniBwdaMgnfTN-FWKbFtisESIBhyw63oeaD2Pmn4iI4DJSOILQQJR38C1EPstHKAPRywCMcDiLHaIcYMCwaAhS7HUWgQi47XwpaWgcUgrSE78gROW6foF6B9D+hAwmYI0upztqKJ6hztagCYRl7ccDQvHYAP-Cfgvx1Ab8GYkhwMB-xaiT8LjqAnASQJoEv1HDrkNw6scaA1ofIbqC6DshWi0nHwgACtocSwbgHEAMyYk+gmQYEPkUkDIZ1AUCbAFJApDPAfB7wB4JSBmJLpQQXwx9iCDMD2DYAXCQ9OwFzDYB8ixmGgM8KpBQjYObwz4NgE+GEBvhSIyrCQCYD5FrqmvYzt7QibmI-a0TAOrExHDB0EmYdAXPZ3wCpBoctvMwIEnt7udRwWxSMO73Sie8qmvnDOnUw1w50guLTHMkXTzKdNzcEfEsr036alC4+4ePuon0HrFcU+VUJyLBB7ZnNdm6fKINDFvBENLI5kCdrqOyjCChYTUM6BOxijaiuUAPMlu5HWCHNsoD0DUcaO1juQUo5ol2BAFtFot7Rpo50S7B9H6irREAK0Wa0NGaivRX+chhaNThNtgxE+bKFEDNEuwJ2uzMCAHBQCBjox5SWMU6LDHrwyB7o9UUaLjHOi9ROYvQYL0Pi5E1OciezOcEuDqAoAkI94b0FWHNigAA)

```
 {
      description: 'patient.careGoal.error.descriptionRequired',
      message: 'patient.careGoal.error.unableToAdd'
    }
```
```
    await act(async () => {
      userEvent.type(screen.getAllByRole('textbox')[0], 'test')
      userEvent.click(screen.getByRole('button', { name: /patient.careGoal.new/i }))
    })
```

Not able to type description.

[AddCareGoalModal.test.tsx] (https://github.com/proful/hospitalrun-frontend/commit/c20c7a9209fdc1cd89e126086141a846812d20b5)

Error gone with below changes

```
```
      userEvent.type(screen.getAllByRole('textbox')[0], 'test')
      userEvent.click(screen.getByRole('button', { name: /patient.careGoal.new/i }))
```

**Comment by nobrayner** [comment]
Closed by #81
